js/ad_manager_vast.js

```diff
@@ -369,16 +369,24 @@ OO.Ads.manager(function(_, $) {
 
       // Save the stream data for use by VideoController
       var streams = {};
+      var streamData = null;
       var linearAd = ad.data.linear;
       if (linearAd && linearAd.mediaFiles) {
         var vastStreams = linearAd.mediaFiles;
-        streams["webm"] = this._extractStreamForType(linearAd.mediaFiles, "webm");
-        if (ad.streamUrl == null) {
-          streams["mp4"] = this._extractStreamForType(linearAd.mediaFiles, "mp4");
-          OO.log("extract m3u8 stream here"); // TODO
+        var videoEncodingsSupported = OO.VIDEO.ENCODING;
+        for (var encoding in videoEncodingsSupported) {
+          streamData = null;
+          if (videoEncodingsSupported.hasOwnProperty(encoding)) {
+            streamData = this._extractStreamForType(vastStreams, videoEncodingsSupported[encoding]);
+          }
+          if (streamData) {
+            streams[videoEncodingsSupported[encoding]] = streamData;
+            if (ad.streamUrl == null || videoEncodingsSupported[encoding] == OO.VIDEO.ENCODING.MP4) {
+              ad.streamUrl = streams[videoEncodingsSupported[encoding]];
```

Comment:
Ad stream url might not be used or only for analytics, not sure, but before it was used to play the video and we were defaulting to mp4, So I didn't want to remove that since I don't know where it is used. was thinking of removing it later when we build the a analytics framework and find out if it is needed or not.